### PR TITLE
WIP (ignore) - refactor(jobsMenu): replace jobs action

### PR DESCRIPTION
**Pull request by weblancaster**
**⚠️  DEPENDS ON https://github.com/dcos/dcos-ui/pull/2999**

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

File: plugins/jobs/src/js/JobsMenuQLSchema.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  type Mutation {
+    runJob(jobId: String!): Job!,
+    updateSchedule(jobId: String, data: String): Job!
+  }
+`;
+
+export const resolvers = {
+  Mutation: {
+    runJob: (parent, args) => {
+      return runJob(args.jobId);
```
Comment (Fabs) [inline plugins/jobs/src/js/JobsMenuQLSchema.js]: Maybe we could have the schuduled calculated here for hasSchedule. 

```
runJob(args.jobId).map((job) => withSchedule(job));
```

Comment (Fabs) [reply plugins/jobs/src/js/JobsMenuQLSchema.js]: Maybe even we need to add this to the queries too, not only on the mutation.

Comment (Fabs) [reply plugins/jobs/src/js/JobsMenuQLSchema.js]: Ideally we could do that but we do not support Field Resolvers yet. 

```
export const resolvers = {
  Fields: {
     schduled: (parent, args) => {
        return withSchedule(parent);
    }
  },
  Mutation: {
    runJob: (parent, args) => {
      return runJob(args.jobId);
    },
    updateSchedule: (parent, args) => {
      return updateSchedule(args.jobId, args.data);
    }
  }
};
```

Comment (Fabs) [reply plugins/jobs/src/js/JobsMenuQLSchema.js]: tip to inspect streams:
.do((v) => console.log(v));

File: plugins/jobs/src/js/JobsToggleSchedule.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  mutation {
+    updateSchedule(jobId: $jobId, data: $data) {
+      jobId
+      data
+    }
+  }
+`;
+
+const enableJobGraphql = job => {
+  return graphqlObservable(enableJobMutation, jobsMenuQLSchema, job);
```
Comment (Fabs) [inline plugins/jobs/src/js/JobsToggleSchedule.js]: For instance, if you had here:
```
updateSchedule(jobId: $jobId, data: $data, searchTerm: $search) {
}
```

The function should be called as:
```graphqlObservable(enableJobMutation, jobsMenuQLSchema, {jobId: 10, data: {}, search: "bananas"});```

File: plugins/jobs/src/js/JobsToggleSchedule.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import { Subject } from "rxjs/Subject";
+import "rxjs/add/operator/combineLatest";
+import "rxjs/add/operator/do";
+import "rxjs/observable/empty";
+
+import { jobsMenuQLSchema } from "./JobsMenuQLSchema";
+
+const enableJobMutation = gql`
+  mutation {
+    updateSchedule(jobId: $jobId, data: $data) {
```
Comment (Fabs) [inline plugins/jobs/src/js/JobsToggleSchedule.js]: This is a js template string, that will require the variables `jobId` and `data` in the context of the graphql execution.


The context is the third argument passed to `graphqlObservable(doc, schema, context)`

File: plugins/jobs/src/js/JobsToggleSchedule.js (hunk 2 above)
Comment (weblancaster) [reply plugins/jobs/src/js/JobsToggleSchedule.js]: Yes that's what I have.

`job` is
```
{
jobId: String,
data: {...}
}
```

**Comment by nLight** [comment]
flicts!

**Comment by weblancaster** [comment]
@nLight I will rebase once @bstavroulakis PR is reviewed and ready to merge because I had to rebase a couple times already, so now I will wait.

@DanielMSchmidt I'm ok moving to `ts` but we need to move @bstavroulakis first.

**Comment by juliangieseke** [comment]
@weblancaster /cc @Fabs is this PR still valid or will it be deprecated by the work fabs is doing? if so, can we close it? ♻️

**Comment by Fabs** [comment]
We can't close it right now, because I am using the information here. It has the label work in progress. I will close once I am done with the work Michael did here on the other PR. Please just ignore it as you would do any other "work in progress" PR.

**Comment by juliangieseke** [comment]
you still can use the information from this branch if the PR is closed - and also look into the PR even if it is closed. I know it has WIP label, but it still distracts us and especially right now the "dont look at WIP stuff" rule doesnt apply :/

**Comment by Fabs** [comment]
@juliangieseke I would like to keep this PR open. So please tell me how do I keep it open and mark it so it does not get seem. 

- The ticket is not in review. So if you are using our JIRA board, you never get here.
- The PR has the WIP (and I know added it to the title)

File: plugins/jobs/src/js/JobsToggleSchedule.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+/* eslint-disable no-unused-vars */
```
Comment (Fabs) [inline plugins/jobs/src/js/JobsToggleSchedule.js]: Note to Self: Part of the graphql is already here.

File: plugins/jobs/src/js/__tests__/JobsMenu-test.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-unused-vars */
```
Comment (Fabs) [inline plugins/jobs/src/js/__tests__/JobsMenu-test.js]: Note to Self: The snapshot testing :D

File: plugins/jobs/src/js/__tests__/JobsRunNow-test.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import { Observable } from "rxjs";
```
Comment (Fabs) [inline plugins/jobs/src/js/__tests__/JobsRunNow-test.js]: Note to self: This test maybe should be ported to when I do delete.

File: plugins/jobs/src/js/data/JobModel.ts [hunk 7]
```diff
@@ -281,13 +546,39 @@ export const resolvers = (): IResolvers => ({
...
+      _context = {}
+    ): Observable<JobRun | null> {
+      return runJob(_args.id);
+    },
+    updateSchedule(
+      _obj = {},
+      _args: GeneralArgs,
+      _context = {}
+    ): Observable<JobRun | null> {
+      return updateSchedule(_args.id, _args.data);
```
Comment (Fabs) [inline plugins/jobs/src/js/data/JobModel.ts]: Note to Self: Mutation here.

**Comment by juliangieseke** [comment]
Its fine to keep it open :) was just trying to clean up a bit 😅

**Comment by nLight** [comment]
@Fabs is it still relevant?

**Comment by Fabs** [comment]
@nLight. No, thanks to remembering. It was noted on the cleanup ticket that we moved out of the sprint.

**Pull request closed without merging**